generator/src/main/java/com/scottlogic/deg/generator/validators/GenerationConfigValidator.java

```diff
@@ -19,6 +23,18 @@ public ValidationResult validateCommandLine(GenerationConfig config) {
             errorMessages.add("RANDOM mode requires max row limit: use -n=<row limit> option");
         }
 
+        if (outputFileAlreadyExists(fileOutputTarget)) {
+            errorMessages.add("Output file already exists, please use a different output filename");
+        }
+
         return validationResult;
     }
+
+    private boolean outputFileAlreadyExists(OutputTarget fileOutputTarget) {
+        if (fileOutputTarget instanceof FileOutputTarget) {
+            Path path = (((FileOutputTarget) fileOutputTarget).getFilePath());
+            return Files.exists(path);
```

Comment:
This becomes less testable, is there a way to inject a `FileSystem` to permit testing here? Or is there another means of testing?

Reply:
functions updated and Files.exists moved into FileOutput class